adminpage/views.py

```diff
@@ -1,3 +1,272 @@
+            name = self.input['name'],
+            key = self.input['key'],
+            place = self.input['place'],
+            description = self.input['description'],
+            pic_url = self.input['picUrl'],
+            start_time = datetime.strptime(self.input['startTime'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8),
+            end_time = datetime.strptime(self.input['endTime'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8),
+            book_start = datetime.strptime(self.input['bookStart'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8),
+            book_end = datetime.strptime(self.input['bookEnd'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8),
+            remain_tickets = self.input['totalTickets'],
+            total_tickets = self.input['totalTickets'],
+            status = self.input['status']
+        )
+        new_activity.save()
+        return new_activity.id
+
+
+class ActivityDetail(APIView):
+
+    def get(self):
+        self.check_input('id')
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User has not logged in.')
+        activity = Activity.get_by_id(self.input['id'])
+        usedTickets = 0
+        for ticket in activity.ticket_set.all():
+            if ticket.status == Ticket.STATUS_USED:
+                usedTickets += 1
+        data = {
+            'name': activity.name,
+            'key': activity.key,
+            'description': activity.description,
+            'startTime': int(time.mktime(activity.start_time.timetuple())),
+            'endTime': int(time.mktime(activity.end_time.timetuple())),
+            'place': activity.place,
+            'bookStart': int(time.mktime(activity.book_start.timetuple())),
+            'bookEnd': int(time.mktime(activity.book_end.timetuple())),
+            'totalTickets': activity.total_tickets,
+            'picUrl': activity.pic_url,
+            'bookedTickets': activity.total_tickets - activity.remain_tickets,
+            'usedTickets': usedTickets,
+            'currentTime': int(time.time()),
+            'status': activity.status
+        }
+        return data
+
+    def post(self):
+        self.check_input('id', 'name', 'place', 'description', 'picUrl',
+                         'startTime', 'endTime', 'bookStart', 'bookEnd',
+                         'totalTickets', 'status')
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User has not logged in.')
+        activity = Activity.get_by_id(self.input['id'])
+        if activity.status == Activity.STATUS_PUBLISHED:
+            if activity.name != self.input['name']:
+                raise LogicError('Cannot modify the name of a published activity ')
+            if activity.place != self.input['place']:
+                raise LogicError('Cannot modify the place of a published activity')
+            if activity.book_start != (datetime.strptime(self.input['bookStart'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8)):
```

Comment:
Hard coding here: You should not add 8 hour to time like this. Instead, you should properly deal with time zone.